recipes/fastjet/meta.yaml

```diff
@@ -0,0 +1,71 @@
+{% set name = "fastjet" %}
+{% set version = "3.4.0.6" %}
+
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/fastjet-{{ version }}.tar.gz
+  sha256: 86dea05ce7a030db9fe14e5bf1c0bc46351b6e439f9b9ef6b14a55ceefcc7b7a
+  patches:
+    - 0001-Don-t-overwrite-external-CXXFLAGS-and-LDFLAGS.patch
+    - 0002-Patch-fastjet-contrib-Makefile.in.patch
+
+build:
+  number: 0
+  skip: true  # [not linux]
+  script: {{ PYTHON }} -m pip install . -vv
+  missing_dso_whitelist:
+  # https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#whitelisting-shared-libraries
+  # All are under lib/python3.*/site-packages/fastjet/_fastjet_core/lib
+    - $RPATH/libfastjet.so.0
+    - $RPATH/libfastjettools.so.0
+    - $RPATH/libsiscone.so.0
+    - $RPATH/libsiscone_spherical.so.0
+    - $RPATH/libfastjetplugins.so.0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - autoconf
+    - automake
+    - libtool
+    - make
+    - patch
+    - swig
+  host:
+    - cgal-cpp  # headers and CGALv5
+    - gmp  # shared library
```

Comment:
Following @xhochy's first review (https://github.com/conda-forge/staged-recipes/pull/21052#pullrequestreview-1887593818) and the patching of `fastjet-contrib/Makefile.in` we are now able to have `gmp` be under `host` while still having it be found by the linker.